api/bot/slackbot/v1/slackbot.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+
+package clutch.bot.slackbot.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/bot/slackbot/v1;slackbotv1";
+
+import "api/v1/annotations.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+service SlackBotAPI {
+  rpc Event(EventRequest) returns (EventResponse) {
+    option (google.api.http) = {
+      post : "/v1/bot/slackbot/event"
+      body : "*"
+    };
+    option (clutch.api.v1.action).type = CREATE;
+  }
+}
+
+// Information on the bot user
+message Bot {
+  // bot id
+  string id = 1;
+  // if the bot is deleted
+  bool deleted = 2;
+  // bot name
+  string name = 3;
+  // timestamp of when the bot app was last updated
+  int64 updated = 4;
+  // unique identifier of the installed Slack application
+  string app_id = 5;
+  // image urls
+  map<string, string> icons = 6;
+  // unique identifier of the workspace where the event occurred
+  string team_id = 7;
+}
+
+// The structure of events vary among types. Full list of event types: https://api.slack.com/events.
+// We will be receiving app_mention events (events that mention the bot) or message.im events (a message posted in a DM
+// with the bot).
+message Event {
+  // the type of event
+  string type = 1;
+  // the user who messaged the bot
+  string user = 2;
+  string bot_id = 3;
+  Bot bot_profile = 4;
+  // the message text
+  string text = 5;
+  // timestamp
+  string ts = 6;
+  // the channel where the event happened
+  string channel = 7;
+  // for DM messages, this will be "im"
+  string channel_type = 8;
+  // when the event was dispatched
+  string event_ts = 9;
+  string client_msg_id = 10;
+  // unique identifier of the workspace where the event occurred
+  string team = 11;
+  // received when a user interacts with a Block Kit component, schema can vary.
+  // https://api.slack.com/reference/block-kit/interactive-components
+  google.protobuf.Value blocks = 12;
+}
+
+// For more details:
+// https://api.slack.com/apis/connections/events-api#the-events-api__receiving-events
+// https://api.slack.com/enterprise/apps/reference#event
+message EventRequest {
+  option (clutch.api.v1.redacted) = true;
+  // verification token to validate the event originated from Slack
+  string token = 1;
+  // if where the event occured is an external shared channel
+  bool is_ext_shared_channel = 2;
+  // unique identifier of the workspace where the event occurred
+  string team_id = 3;
+  // unique identifier of the installed Slack application
+  string api_app_id = 4;
```

Comment:
is this scoped to only the API? I think the app ID is shared everywhere

Reply:
the slack API sends us this field to our endpoint, and it's only on the outer layer of the request
example
```
{
    "token": "XXYYZZ",
    "team_id": "TXXXXXXXX",
    "api_app_id": "AXXXXXXXXX",
    "event": {
            "type": "name_of_event",
            "event_ts": "1234567890.123456",
            "user": "UXXXXXXX1",
            ...
    },
```